Set up my project developing enviroment

Xhz0729 opened the pull request:
**Summary:** 
This pull request introduces several foundational features to the project, including the setup of the frontend, backend, and database, as well as initial user authentication routes.

**What's been done:**

- Connected the project with GitHub repository for version control and collaboration.
- Set up React frontend using Vite for fast development and optimized builds.
- Configured Tailwind CSS for utility-first styling.
- Created the header component to provide consistent navigation across pages.
- Implemented React Router to enable client-side routing.
- Developed login and registration pages to allow user authentication.
- Set up an Express backend to handle server-side logic.
- Created MongoDB database for data persistence.
- Established Mongoose connection to interact with MongoDB.

Xhz0729 committed:
add DreamStay logo and DreamStay company name

Xhz0729 committed:
finish my header, add search area, menu bar and user icon

Xhz0729 committed:
configure my primary color

Xhz0729 committed:
install react router package

Xhz0729 committed:
extract header as a component and integrate with React Router

Xhz0729 committed:
move my HeaderPage component into the src folder and render it in Layout component

Xhz0729 committed:
add a LoginPage component and render it in App.jsx with styling

Xhz0729 committed:
add RegisterPage component and render it on App.jsx

Xhz0729 committed:
fix typo, change 'Login' to 'Register

Xhz0729 committed:
add a gitignore file to help untrack .env file and node modules

Xhz0729 committed:
set up express backend

Xhz0729 committed:
add User schema

Xhz0729 committed:
install nodemon, concurrently and bcrypt packages

Xhz0729 committed:
add register route

use mongoose to connect with database, use .env and dotenv to configure db url and use post method to create new user

Xhz0729 committed:
fix bugs by changing all property names to camelCase

Xhz0729 committed:
handle my form input changes and handle my form submit

Xhz0729 committed:
install axios package

ericadev commented on server/server.js:
```diff
@@ -0,0 +1,38 @@
+
+mongoose.connect(process.env.MONGO_URL);
+
+app.get("/", (req, res) => {
```
You could clean up a little by removing this unnecessary code

ericadev commented on client/src/HeaderPage.jsx:
```diff
@@ -0,0 +1,89 @@
+import React from "react";
```
Maybe just me but I got confused by this component name, since it has Page in the name I was thinking it was a page but it's just a header.

Xhz0729 replied:
This is a good point, I think Header would be better as the component name.

ericadev commented on client/src/pages/RegisterPage.jsx:
```diff
@@ -0,0 +1,74 @@
+        email,
+        password,
+      });
+      alert("Registration successful. Now you can log in");
```
Is there a better way than using an alert for this?

Xhz0729 replied:
I'm still working on the RegisterPage. The alert is a temporary solution to test that the API call is functioning correctly. Moving forward, I plan to display a message element on the page, which will provide a more consistent and user-friendly experience.

Xhz0729 replied:
@ericadev I'm still working on the RegisterPage. The alert is a temporary solution to test that the API call is functioning correctly. Moving forward, I plan to display a message element on the page, which will provide a more consistent and user-friendly experience.

Xhz0729 committed:
add a readme with project overview and link to trello

Xhz0729 committed:
add type submit to my button

Xhz0729 committed:
Await the bcrypt hash password

Xhz0729 committed:
add the login route

Xhz0729 committed:
add login logic

add formData state to track user input and connect the component with backend route

ChasVanDav commented:
Your set up so far looks great! I am looking forward to helping you create this awesome app!
